sys/cpp_new_delete/new_delete.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+  Copyright (c) 2014 Arduino.  All right reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include <stdlib.h>
+
+void *operator new(size_t size) {
+  return malloc(size);
+}
+
+void *operator new[](size_t size) {
+  return malloc(size);
+}
+
+void * operator new(size_t size, void * ptr) noexcept {
+  (void)size;
+  return ptr;
+}
```

Comment:
I don't quite get this one. I cannot find an API doc, but the arguments do not seem to make much sense unless this expected to behave somewhat like realloc

Reply:
According to https://www.cplusplus.com/reference/new/operator%20new/
```
void* operator new (std::size_t size, void* ptr) noexcept;
```
is the `placement` function which simply returns `ptr` and doesn't allocate storage:
```
ptr 
       A pointer to an already-allocated memory block of the proper size.
       If called by a new-expression, the object is initialized (or constructed) at this location.
```

Reply:
The counterpart is
```
void operator delete (void* ptr, void* voidptr2) noexcept;
```

Reply:
Found an good example how it could be used at [cppreference.com](https://en.cppreference.com/w/cpp/language/new)
```
// within any block scope...
{
    alignas(T) unsigned char buf[sizeof(T)];
    // Statically allocate the storage with automatic storage duration
    // which is large enough for any object of type `T`.
    T* tptr = new(buf) T; // Construct a `T` object, placing it directly into your 
                          // pre-allocated storage at memory address `buf`.
    tptr->~T();           // You must **manually** call the object's destructor
                          // if its side effects is depended by the program.
}                         // Leaving this block scope automatically deallocates `buf`.
```